### PR TITLE
add license info of commandz, #6324

### PR DESCRIPTION
**Pull request by asdf168ghjkl**
ref : https://github.com/EtienneLem/commandz/blob/master/MIT-LICENSE

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
Please separate the reference issue by a comma, like this:
`add license info of commandz, #6324`

File: ajax/libs/commandz/package.json [hunk 1]
```diff
@@ -28,5 +28,6 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/EtienneLem/commandz"
-  }
+  },
+  "liencese": "MIT"
```
Comment (PeterDaveHello) [inline ajax/libs/commandz/package.json]: Wrong spelling.

**Comment by asdf168ghjkl** [comment]
corrected

**Comment by PeterDaveHello** [comment]
@asdf168ghjkl you'll need to squash them into only one commit so that we can merge it, because there is no reason to add a license info with multiple commits, all branch should be clean and clear before been merged, so that we can have a cleaner master branch.

Or you can also close this PR and open a new one if you wanna do it easily instead.

**Comment by asdf168ghjkl** [comment]
@PeterDaveHello can you teach me how to squash them, please ?

**Comment by PeterDaveHello** [comment]
Of course, happy to see you wanna learn it, I'll let @Amomo help you.

**Comment by Amomo** [comment]
Hi @asdf168ghjkl 
Since you can't modify the commit message or delete your old commit in your browser.

If you want to squash your commits, you need to do it locally.
Clone your fork cdnjs repo, and pull your branch "patch-2" to local, so you can squash these commits by git commands. ([This](https://help.github.com/articles/about-git-rebase/#rebasing-commits-against-a-point-in-time) could help you to use git commands)

There are 2 methods to clone your fork cdnjs repo, you can clone it directly(but it take time), or you can study our [sparseCheckout.md](https://github.com/cdnjs/cdnjs/blob/master/sparseCheckout.md) to pull your own repo then pull the branch(a little hard).

You may run into some problems in the second method, if you are't familiar with git.
It's welcome to ask us if you need any help! :-)

**Comment by asdf168ghjkl** [comment]
what should I do if it says "sparse checkout leaves no entry on working directory"?

**Comment by Amomo** [comment]
@asdf168ghjkl Have you create a file named `"sparse-checkout"` in .git/info/ ?

**Comment by asdf168ghjkl** [comment]
@Amomo yes

**Comment by Amomo** [comment]
Hi @asdf168ghjkl sorry for late reply.
I am not sure which situation you met, could you tell me what steps you had done?

**Comment by asdf168ghjkl** [comment]
@Amomo 
These are my commands
$ git init cdnjs 
$ cd cdnjs
$ git config core.sparseCheckout true
$ echo '/ajax/libs/commandz/*' >> .git/info/sparse-checkout$ git remote add origin git://github.com/asdf168ghjkl/cdnjs.git
$ git pull origin master --depth 10
$ git pull
$ git checkout origin/patch-2

then it shows "error : Sparse checkout leaves no entry on working directory"

**Comment by PeterDaveHello** [comment]
@asdf168ghjkl try:
`
$ git fetch origin patch-2:patch-2
$ git checkout patch-2
`

**Comment by asdf168ghjkl** [comment]
@PeterDaveHello it shows the same thing

**Comment by PeterDaveHello** [comment]
@asdf168ghjkl would you please show me your result of `git log`? 
Due to you are trying to modify an old commit/PR, `--depth 10` may not be enough, would you please try `--depth 600` instead `--depth 10` and see if it works? Thanks.

**Comment by asdf168ghjkl** [comment]
@PeterDaveHello 
$ git log
fatal: your current branch 'master' does not have any commits yet.

and when i enter $ git pull origin master --depth 600
it shows
remote: total 0
from git://..............
 * branch           master   -> FETCH_HEAD
error : Sparse checkout leaves no entry on working directory

**Comment by PeterDaveHello** [comment]
@asdf168ghjkl So I thought you didn't pull the repository with success, am I right?

Actually I did the same process with this steps and had no problem:

```sh
user@ubuntu ~ $ git init cdnjs
Initialized empty Git repository in /home/user/cdnjs/.git/
user@ubuntu ~ $ cd cdnjs
user@ubuntu ~/cdnjs $ git config core.sparseCheckout true
user@ubuntu ~/cdnjs $ echo '/ajax/libs/commandz/*' >> .git/info/sparse-checkout
user@ubuntu ~/cdnjs $ git remote add origin git://github.com/asdf168ghjkl/cdnjs.git

user@ubuntu ~/cdnjs $ git pull origin master --depth 600
remote: Counting objects: 377117, done.
remote: Compressing objects: 100% (213243/213243), done.
remote: Total 377117 (delta 158184), reused 368252 (delta 156827), pack-reused 0
Receiving objects: 100% (377117/377117), 1003.54 MiB | 1.24 MiB/s, done.
Resolving deltas: 100% (158184/158184), done.
From git://github.com/asdf168ghjkl/cdnjs
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> origin/master
```

Would you please try again with a clean space and these commands?

```sh
git init cdnjs 
cd cdnjs
git config core.sparseCheckout true
echo '/ajax/libs/commandz/*' >> .git/info/sparse-checkout
git remote add origin git://github.com/asdf168ghjkl/cdnjs.git
git pull origin master --depth 10
```
I noticed that you cloned cdnjs from your fork(asdf168ghjkl/cdnjs), not our main repo(cdnjs/cdnjs), so `--depth 10` will be very enough for you, no need to use 600 :smile: 

Thanks!

**Comment by asdf168ghjkl** [comment]
I can pull the repository now
then what should I do to squash them?
Thanks!

**Comment by Amomo** [comment]
Hi @asdf168ghjkl , to squash your latest two commit, you can use 
```
git rebase -i HEAD~~
``` 

Then, Git will launch your editor and open a file that details the commits in the range you've chosen. 
That file looks something like this:

```sh
pick d743cab add license info of commandz #6324
pick 54997a4 Update package.json

# Rebase 7bccf77..54997a4 onto 7bccf77 (2 command(s))
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
#  ...
```
Each line lists a command (by default, pick), the commit SHA, and the commit message. 

Command `squash` or `fixup` will meld the commit into previous commit.
You don't need commit message about "Update package.json", so I suggest you to use `fixup`.
modify `pick 54997a4 Update package.json` to 
```
fixup 54997a4 Update package.json
```

After write & quit the file, you will see two commits combine into one. [(more detail)](https://help.github.com/articles/about-git-rebase/#rebasing-commits-against-a-point-in-time)

If everything is all right, next we need to make your commit ahead of our master branch!

**Comment by PeterDaveHello** [comment]
> If everything is all right, next we are going to make your commit ahead of our master branch, because this commit is a little bit too old.
Nope, its parent was old, should rebase on latest master branch.

**Comment by Amomo** [comment]
oh yes, **latest** master branch. :sparkles:

**Comment by PeterDaveHello** [comment]
So @Amomo you should say how to do it.

**Comment by PeterDaveHello** [comment]
@Amomo you should also talk about the shallow clone and sparse-checkout.

**Comment by asdf168ghjkl** [comment]
@Amomo 
Am I right?
$ git checkout patch-2
$ git rebase -i HEAD~~
thanks

**Comment by Amomo** [comment]
@asdf168ghjkl see your `git log`
Do you see two commits combine into one?

**Comment by asdf168ghjkl** [comment]
@Amomo 
it shows:
Merge pull request #6343 from kirjs/master
[new][author] Add react-highcharts@5.0.6

**Comment by Amomo** [comment]
It seems that two of your commits meld into the previous commit.
Did you change two `pick` into `fixup`???

Here only need to change `one` line to fixup, 
make "Update package.json" meld to "add license info of commandz #6324".

**Comment by Amomo** [comment]
Let's go back again. try
```
$ git reset HEAD^
$ git status
```
check the message, then
```
$ git pull origin patch-2
```

**Comment by asdf168ghjkl** [comment]
@Amomo 
So what should i left in # Commands:?
# p, pick = use commit
# f, fixup = like "squash", but discard this commit's log message
??
thanks

**Comment by Amomo** [comment]
make
```
pick d743cab add license info of commandz #6324
pick 54997a4 Update package.json
```
to
```
pick d743cab add license info of commandz #6324
fixup 54997a4 Update package.json
```

**Comment by asdf168ghjkl** [comment]
@Amomo 
Am I done?
$ git log --oneline --graph
* 41b77fa add license info of commandz #6324
*   7bccf77 Merge pull request #6343 from kirjs/master
|\
| * 04991ed Add react-highcharts@5.0.6
* |   97e654e Merge pull request #6278 from Amomo/jquery-treetable
|\ \
| * | 9641696 add jquery-treetable@3.2.0 with git auto-update
* | |   aa2a905 Merge pull request #6150 from Amomo/checklist-model
|\ \ \
| * | | 1084450 add checklist-model@0.6.0 with git auto-update
| |/ /
* | |   364666b Merge pull request #6355 from pvnr0082t/pvnr0082t-architect-license
|\ \ \
| * | | e0cf84e add missing license of architect
* | | |   340bbb1 Merge pull request #6353 from limitimil/limitimil-gumby-license2
|\ \ \ \
| * | | | c70ce03 add missing license info of gumby
* | | | | 2ae909b add web-animations v2.1.4
* | | | | e1977e2 add danialfarid-angular-file-upload v10.1.1/10.1.2
* | | | | 9837d2b add graphiql v0.4.3/0.4.4
* | | | | 1da05cb add vega v2.4.0
* | | | | 05025c3 add typescript v1.7.3
* | | | | bb71429 add simple-statistics v1.0.1

**Comment by Amomo** [comment]
@asdf168ghjkl yes, seems that you get the new commit (new commit SHA `41b77fa`).

Then, you have to fetch to our latest master branch.
```sh
$ git remote add upstream git://github.com/cdnjs/cdnjs.git
$ git fetch upstream master
```
After that,
```sh
$ git rebase upstream/master
```
try `$ git log` again. :-)
You will see your commit is ahead of our latest master branch.

**Comment by Amomo** [comment]
If everything is fine, **force push** your current branch to your GitHub repo.
```sh
$ git push -f origin patch-2
```
Finally, you will see some change in this page.
The number of your commit in this PR changed into one.
And, your commit is newer than our latest master branch.

**Comment by asdf168ghjkl** [comment]
$ git log
commit 2d5fe797d5869ad4d43659d7c4d37e8c869cdfa3
Author: asdf168ghjkl <asdf168ghjkl@gmail.com>
Date:   Tue Dec 1 22:33:12 2015 +0800

    add license info of commandz #6324

    ref : https://github.com/EtienneLem/commandz/blob/master/MIT-LICENSE

commit 45254a094881d6d824936adf4e1e35dc91436007
Author: PeterDaveHello <hsu@peterdavehello.org>
Date:   Fri Dec 18 15:51:57 2015 +0800

    add emojione v2.0.0

commit 4d35d1c29e4da99ec41614cf091dea2c7ea60779
Author: Peter Dave Hello <hsu@peterdavehello.org>
Date:   Fri Dec 18 15:15:11 2015 +0800

    add ol3 v3.12.1

commit 73db3614331860b5232067f5fc7d3cdc56a8c930
Author: Peter Dave Hello <hsu@peterdavehello.org>
Date:   Fri Dec 18 14:59:14 2015 +0800

OK?

**Comment by Amomo** [comment]
@asdf168ghjkl yes, then you can use force push.

**Comment by asdf168ghjkl** [comment]
@Amomo 
it shows this
$ git push -f origin patch-2
error: src refspec patch-2 does not match any.
error: failed to push some refs to 'git://github.com/asdf168ghjkl/cdnjs.git'

**Comment by Amomo** [comment]
Hmmm, maybe it is a ssh connection problem.
Let's set the remote url to **https**.
```sh
$ git remote set-url origin https://github.com/asdf168ghjkl/cdnjs.git
```
Then, try to force push again.

**Comment by asdf168ghjkl** [comment]
@Amomo 
It shows the same.

**Comment by PeterDaveHello** [comment]
@asdf168ghjkl what about this:
`git push origin patch-2:test-patch-2`

Will it work?

**Comment by asdf168ghjkl** [comment]
@PeterDaveHello 
$ git push origin patch-2:test-patch-2
error: src refspec patch-2 does not match any.
error: failed to push some refs to 'https://github.com/asdf168ghjkl/cdnjs.git'

**Comment by PeterDaveHello** [comment]
@asdf168ghjkl could your shoule me your result of `git branch`?
I guess you are not at `patch-2` but maybe `origin/patch-2`

**Comment by asdf168ghjkl** [comment]
@PeterDaveHello 
$ git branch
* master

**Comment by PeterDaveHello** [comment]
So what's the result of the actions this comment?

https://github.com/cdnjs/cdnjs/pull/6361#issuecomment-164127056
```sh
$ git fetch origin patch-2:patch-2
$ git checkout patch-2
```

I thought you missed something.

**Comment by asdf168ghjkl** [comment]
@PeterDaveHello 
these are my commands:

git init cdnjs 
cd cdnjs
git config core.sparseCheckout true
echo '/ajax/libs/commandz/*' >> .git/info/sparse-checkout
git remote add origin git://github.com/asdf168ghjkl/cdnjs.git
git pull origin master --depth 10
git pull orign patch-2
git rebase -i HEAD~~

**Comment by PeterDaveHello** [comment]
Okay but I need the response of the commands, so that I can confirm the result, and you should not use git protocol, see [Which remote URL should I use?](https://help.github.com/articles/which-remote-url-should-i-use/)

**Comment by PeterDaveHello** [comment]
cc @cdnjs/intern to help review this PR

**Comment by asdf168ghjkl** [comment]
@PeterDaveHello 
these are my results

裕衡@puling-pc MINGW64 /d/Downloads/cdnjs (patch-2)
$ git push -f origin patch-2
fatal: remote error:
  You can't push to git://github.com/asdf168ghjkl/cdnjs.git
  Use https://github.com/asdf168ghjkl/cdnjs.git

裕衡@puling-pc MINGW64 /d/Downloads/cdnjs (patch-2)
$ git remote set-url origin https://github.com/asdf168ghjkl/cdnjs.git

裕衡@puling-pc MINGW64 /d/Downloads/cdnjs (patch-2)
$ git push -f origin patch-2
Counting objects: 16841, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (8789/8789), done.
Writing objects: 100% (16841/16841), 59.29 MiB | 309.00 KiB/s, done.
Total 16841 (delta 7369), reused 16832 (delta 7364)
To https://github.com/asdf168ghjkl/cdnjs.git
 + 54997a4...fb66f2b patch-2 -> patch-2 (forced update)

**Comment by PeterDaveHello** [comment]
Looks nice now! Maybe just lost some steps in the middle, so without that misses, you now know how to do this!

**Pull request merged**